mount/fs.go

```diff
@@ -55,6 +57,9 @@ func (f *FSMount) Serialize() *url.URL {
 	if st, err := fs.Stat(f.FS, f.Path); err != nil {
 		u.Host = "irrecoverable"
 	} else {
+		q := u.Query()
+		q.Set(path, f.Path)
+		u.RawQuery = q.Encode()
 		u.Host = st.Name()
```

Comment:
Basically what you discovered here is that `st.Name() != f.Path`. I was going on that assumption, but if that's not the case, let's just set the path in the URL host, instead of `st.Name()`

Reply:
Yeah, but then the URL parsing then fails.  Is there a way to put filepaths in URL hostnames ?